src/main/java/com/waverleysoftware/listemers/TestListener.java

```diff
@@ -1,31 +1,41 @@
 package com.waverleysoftware.listemers;
 
 import lombok.extern.slf4j.Slf4j;
+import org.assertj.core.api.SoftAssertions;
 import org.testng.IInvokedMethod;
 import org.testng.IInvokedMethodListener;
 import org.testng.ITestResult;
+import org.testng.internal.TestResult;
 
-@SuppressWarnings("JavadocType")
+@SuppressWarnings({"JavadocType", "unchecked"})
```

Comment:
avoid using `unchecked` on class level... annotate exact method, where it's required... however, I believe that type casting warning will go away, when you specify an exact generic type